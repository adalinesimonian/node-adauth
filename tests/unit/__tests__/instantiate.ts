import ADAuth from '../../../lib/adauth'

describe('Using constructor', () => {
  test('Throws with missing options', () => {
    expect(() => new (ADAuth as any)()).toThrowErrorMatchingSnapshot()
  })

  test('Throws with missing URL', () => {
    expect(
      () =>
        new ADAuth({
          domainDN: 'dc=example,dc=com',
        } as any)
    ).toThrowErrorMatchingSnapshot()
  })

  test('Throws with missing domain DN', () => {
    expect(
      () =>
        new ADAuth({
          url: 'ldap://ldap.forumsys.com:389',
        } as any)
    ).toThrowErrorMatchingSnapshot()
  })

  test('Creates instance with valid options', () => {
    expect(
      new ADAuth({
        url: 'ldap://ldap.forumsys.com:389',
        domainDN: 'dc=example,dc=com',
      })
    ).toBeInstanceOf(ADAuth)
  })
})

describe('Using static create method', () => {
  test('Rejects with missing options', () => {
    expect((ADAuth as any).create()).rejects.toMatchSnapshot()
  })

  test('Rejects with missing URL', () => {
    expect(
      ADAuth.create({
        domainDN: 'dc=example,dc=com',
      } as any)
    ).rejects.toMatchSnapshot()
  })

  test('Rejects with missing domain DN', () => {
    expect(
      ADAuth.create({
        url: 'ldap://ldap.forumsys.com:389',
      } as any)
    ).rejects.toMatchSnapshot()
  })

  test('Creates instance with valid options', async () => {
    const auth = await ADAuth.create({
      url: 'ldap://ldap.forumsys.com:389',
      domainDN: 'dc=example,dc=com',
    })
    expect(auth).toBeInstanceOf(ADAuth)
    await auth.dispose()
  })
})
