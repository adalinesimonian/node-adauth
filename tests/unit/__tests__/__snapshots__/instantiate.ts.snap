// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using constructor Throws with missing URL 1`] = `"AD domain controller LDAP URL not defined (options.url)"`;

exports[`Using constructor Throws with missing domain DN 1`] = `"Domain DN not defined (options.domainDN)"`;

exports[`Using constructor Throws with missing options 1`] = `"Options not provided"`;

exports[`Using static create method Rejects with missing URL 1`] = `[AssertionError: AD domain controller LDAP URL not defined (options.url)]`;

exports[`Using static create method Rejects with missing domain DN 1`] = `[AssertionError: Domain DN not defined (options.domainDN)]`;

exports[`Using static create method Rejects with missing options 1`] = `[AssertionError: Options not provided]`;
